hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsReadFooterMetrics.java

```diff
@@ -17,533 +17,348 @@
-    readFooterMetric += getReadFooterMetrics(avgNonparquetReadFooterMetrics);
-  }
-  return readFooterMetric;
-}
 
+    private synchronized void handleFurtherRead(CheckFileType checkFileType, int len) {
+        if (checkFileType.getCollectLenMetrics() && checkFileType.getFileType() != null) {
+            FileType fileType = checkFileType.getFileType();
+            updateMetricValue(fileType, READ_LEN_REQUESTED, len);
+            incrementMetricValue(fileType, READ_COUNT);
+        }
+    }
+
+    private synchronized void updateMetricsData(CheckFileType checkFileType, int len, long contentLength) {
+        long sizeReadByFirstRead = Long.parseLong(checkFileType.getSizeReadByFirstRead().split("_")[0]);
+        long firstOffsetDiff = Long.parseLong(checkFileType.getSizeReadByFirstRead().split("_")[1]);
+        long secondOffsetDiff = Long.parseLong(checkFileType.getOffsetDiffBetweenFirstAndSecondRead().split("_")[1]);
+        FileType fileType = checkFileType.getFileType();
+
+        updateMetricValue(fileType, READ_LEN_REQUESTED, len + sizeReadByFirstRead);
+        updateMetricValue(fileType, FILE_LENGTH, contentLength);
+        updateMetricValue(fileType, SIZE_READ_BY_FIRST_READ, sizeReadByFirstRead);
+        updateMetricValue(fileType, OFFSET_DIFF_BETWEEN_FIRST_AND_SECOND_READ, len);
+        updateMetricValue(fileType, FIRST_OFFSET_DIFF, firstOffsetDiff);
+        updateMetricValue(fileType, SECOND_OFFSET_DIFF, secondOffsetDiff);
+        incrementMetricValue(fileType, TOTAL_FILES);
+    }
 
-  @Override
-  public String toString() {
-    Map<String, AbfsReadFooterMetrics> metricsMap = getMetricsMap();
-    List<AbfsReadFooterMetrics> readFooterMetricsList = new ArrayList<>();
-    if (metricsMap != null && !(metricsMap.isEmpty())) {
-      checkIsParquet(metricsMap);
-      updateLenRequested(metricsMap);
-      for (Map.Entry<String, AbfsReadFooterMetrics> entry : metricsMap.entrySet()) {
-        AbfsReadFooterMetrics abfsReadFooterMetrics = entry.getValue();
-        if (abfsReadFooterMetrics.getCollectMetrics()) {
-          readFooterMetricsList.add(entry.getValue());
+    /**
+     * Returns the read footer metrics for a given file type.
+     *
+     * @param fileType the type of the file
+     * @return the read footer metrics as a string
+     */
+    public String getReadFooterMetrics(FileType fileType) {
+        StringBuilder readFooterMetric = new StringBuilder();
+        appendMetrics(readFooterMetric, fileType);
+        return readFooterMetric.toString();
+    }
+
+    private void appendMetrics(StringBuilder metricBuilder, FileType fileType) {
+        long totalFiles = getMetricValue(fileType, TOTAL_FILES);
+        long readCount = getMetricValue(fileType, READ_COUNT);
+        if (totalFiles <= 0 || readCount <= 0) {
+            return;
+        }
+
+        String sizeReadByFirstRead = String.format("%.3f", getMetricValue(fileType, SIZE_READ_BY_FIRST_READ) / (double) totalFiles);
```

Comment:
String.format can become a method call as :- private String formatMetric(long value, long divisor) {
    return String.format("%.3f", value / (double) divisor);
}

Reply:
created a StringUtils class and added this method.